service/db_test.go

```diff
@@ -255,3 +255,12 @@ func TestUpdateUserRow(t *testing.T) {
 		}
 	}
 }
+
+func TestInsertNewSecret(t *testing.T) {
+	counter := 3
+	for counter != 0 {
+		err := insertNewSecret()
+		assert.Nil(t, err)
+		counter--
+	}
```

Comment:
This specific test should query the DB to check if a new secret was generated. You should look up DAO or Data Access Object. You can add up to the `db.go` just  to query the db.

Reply:
Write a func inside your db.go to query the database to check if an entry was added in the table.

Call the func inside your test.

Your db.go is your DAO.

Reply:
DAO is reponsible for talking to the db.

your db.go does this particular job. You actually made a DAO.